cocotb/regression.py

```diff
@@ -127,17 +127,21 @@ def __init__(self, dut: SimHandle, tests: Iterable[Test]):
 
         # Test Discovery
         ####################
-        self._queue = []
         for test in tests:
             self.log.info(f"Found test {test.__module__}.{test.__qualname__}")
-            self._queue.append(test)
-        self.ntests = len(self._queue)
+        self._queue = list(tests)
 
         if not self._queue:
             self.log.warning("No tests were discovered")
+            return
 
         self._queue.sort(key=lambda test: (test.stage, test._id))
 
+    @property
+    def ntests(self) -> int:
+        """Gets the number of tests in this regression."""
```

Comment:
```suggestion
        """Get the number of tests in this regression."""
```

Reply:
Thanks for the suggestion. I kept it as-is to be in line with the rest of the file, which uses "XXXs" as well.